code/hsec-core/src/Security/Advisories/Core/Advisory.hs

```diff
@@ -44,6 +47,13 @@ data Advisory = Advisory
   }
   deriving stock (Show)
 
+data Ecosystem = Hackage | GHC GHCComponent
```

Comment:
I think it should be `Affected`, or be the type of `Affected.affectedPackage`, which might be renamed `Affected.affectedComponent`.

Otherwise we might be able to name a package (e.g. `aeson`) while being in `GHCi`.

Reply:
Though we can already mismatch a package with the name from the path. So it seems like we need an extra validation to ensure the package/ghc component from the path match at least one of the affected.

Well I assumed a given advisory would apply to a single `Ecosystem`, but perhaps this is not true. In that case, I think we need to update the affected schema to have either a `package` or a `component` key to parse a single `Affected.affectedEcosystem` and remove the `affectedPackage`.

```haskell
data AffectedEcosystem = Hackage PackageName | GHC GHCComponent
```

What do you think?

Reply:
It seems the right way to do it, thanks.

Reply:
This is now implemented, could you please take another look?